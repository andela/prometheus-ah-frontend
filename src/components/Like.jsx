import React from 'react';
import PropTypes from 'prop-types';

/**
 * function Like
 * @returns {Object} Like component
 */
const Like = ({
  iconClassName, spanClassName, count, ...rest
}) => (
  <i className={iconClassName} {...rest}>
    <span className={spanClassName}>
      {' '}
      {count}
    </span>
  </i>
);
Like.propTypes = {
  iconClassName: PropTypes.string,
  spanClassName: PropTypes.string,
  count: PropTypes.number
};

Like.defaultProps = {
  iconClassName: '',
  spanClassName: ''
};

export default Like;
