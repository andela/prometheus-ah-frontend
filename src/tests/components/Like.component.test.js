import React from 'react';
import { shallow } from 'enzyme';
import Like from '../../components/Like';

let wrapper;

describe('Like', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Like />
    );
  });

  test('should correctly Like component ', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
