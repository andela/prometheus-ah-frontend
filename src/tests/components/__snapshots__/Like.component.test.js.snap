// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like should correctly Like component  1`] = `
<i
  className=""
>
  <span
    className=""
  >
     
  </span>
</i>
`;
